src/Cli/dotnet/CommonOptions.cs

```diff
@@ -59,7 +59,7 @@ internal static class CommonOptions
             {
                 Argument = new Argument<string>(CommonLocalizableStrings.ConfigurationArgumentName)
                     .AddSuggestions(Suggest.ConfigurationsFromProjectFileOrDefaults().ToArray())
-            }.ForwardAsSingle(o => $"-property:Configuration={o}");
+            }.ForwardAsSingle(o => o.StartsWith("lp:") ? $"-c{o}" : $"-property:Configuration={o}");
```

Comment:
Are there any other switches that we missed? -clp isn't listed on our CLI docs so now I'm worried there are others we need to account for. Should this one be added to our dotnet build docs?

Reply:
For context, this issue is a result of the way System.CommandLine handles options that are prefixes of each other, for example in this case `-c` (short form of `--configuration`) and `-clp`. System.CommandLine parses `-clp:blah` as the configuration option with the input `lp:blah`, which we obviously don't want because -clp is a MSBuild command line switch, so we have to write a special case for this. The only other one I've found prior to this is `-p` and `-property`. 

I'll audit the msbuild command line options to see if there are any other cases where one of those overlaps with one of the .NET options.

Reply:
See the discussion of this issue on a previous PR: https://github.com/dotnet/sdk/pull/14664/files#r529782097. Looking at the MSBuild command line reference, it seems like there are ~5 other instances where this behavior could be a problem and require us to add this kind of workaround to allow parsing those msbuild options. 

@jonsequitur in the thread I linked you pointed out that System.CommandLine is following the GNU spec but MSBuild has different behavior. Would you be willing to add a option to opt into the MSBuild style behavior so we can avoid having to have all of these workarounds? I previously made a PR to do this so it would be simple to reopen it and make the new behavior opt-in: https://github.com/dotnet/command-line-api/pull/1093. cc @wli3 @dsplaisted @marcpopMSFT